From dd9b996eeea5f39715e3ab9469dd343933480d3f Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:17:30 +0200
Subject: [PATCH 19/50] AnnaThermostat: add support for
 domestic_hot_water_state

Parse domestic_hot_water_state and publish it as an on/off switch sensor.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 6fd398866..225636577 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1167,6 +1167,14 @@ void CAnnaThermostat::GetMeterDetails()
 					SendSwitch(applianceIDhex, 6, batterypercentage, isOn, 0, ApplianceName + " " + sname, m_Name);
 				}
 			}
+			else if (sname == "domestic_hot_water_state")
+			{
+				if (!tmpstr.empty())
+				{
+					bool isOn = (strcmp(tmpstr.c_str(), "on") == 0);
+					SendSwitch(applianceIDhex, 7, batterypercentage, isOn, 0, ApplianceName + " " + sname, m_Name);
+				}
+			}
             else if (sname == "central_heater_water_pressure")
             {
                 if (!tmpstr.empty())
-- 
2.51.0

